dask/dataframe/core.py

```diff
@@ -6094,8 +6094,7 @@ def repartition_freq(df, freq=None):
     if not len(divisions):
         divisions = [df.divisions[0], df.divisions[-1]]
     else:
-        if divisions[-1] != df.divisions[-1]:
-            divisions.append(df.divisions[-1])
+        divisions.append(df.divisions[-1])
```

Comment:
Doesn't this mean that there could potentially be two identical divisions which would result in an empty partition?

Reply:
Yes it _could_ be that the last two divisions are identical, but this is sometimes what you want. E.g., in  https://github.com/dask/dask/issues/7139 your divisions should be `[pd.Timestamp(2020, 1, 1), pd.Timestamp(2020, 1, 2), pd.Timestamp(2020, 1, 2)]`, if my understanding of divisions is correct. 

Could you maybe give a counterexample of how/when this goes wrong? I cannot come up with one :-).

Reply:
So, if your very last division (i.e., the maximum of in the index) is in the `pd.date_range` (i.e., `divisions[-1] == df.divisions[-1]`) then it should be double.

If this is not the case, it shouldn't be double and this is exactly what happens. I think.

Reply:
I think it's OK, as long as we know that the last partition has only the one value. Our requirement is that divisions has the minimum of each partition and the maximum of the final partition. Is that the case here?